** Define a Debian-specific Emacs flavor.
   Provided by: Rob Browning <rlb@defaultvalue.org>

---
 lisp/startup.el |   12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

--- a/lisp/startup.el
+++ b/lisp/startup.el
@@ -375,6 +375,10 @@
 
 (defvar normal-top-level-add-subdirs-inode-list nil)
 
+(defconst debian-emacs-flavor 'emacs-snapshot
+  "A symbol representing the particular debian flavor of emacs running.
+Something like 'emacs20, 'xemacs20, etc.")
+
 (defvar no-blinking-cursor nil)
 
 (defvar default-frame-background-mode)
@@ -956,7 +960,13 @@
     ;; should check init-file-user instead, since that is already set.
     ;; See cus-edit.el for an example.
     (if site-run-file
-	(load site-run-file t t))
+	(progn
+	  ;; Load all the debian package snippets.
+	  ;; It's in here because we want -q to kill it too.
+	  (if (load "debian-startup" t t nil)
+	      (debian-startup debian-emacs-flavor))
+	  ;; Now the normal site file...
+	  (load site-run-file t t nil)))
 
     ;; Sites should not disable this.  Only individuals should disable
     ;; the startup screen.
